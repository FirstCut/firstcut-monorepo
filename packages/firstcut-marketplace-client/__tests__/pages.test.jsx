
import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import wait from 'waait';
import {
  ContactPage,
  GET_PROJECT_QUERY,
  ExploreMarketplacePage,
  GET_PROJECTS_QUERY,
} from '../src/pages';
import Loading from '../src/components/loading';

configure({ adapter: new Adapter() });

const testProject = {
  _id: '1',
  title: 'Title',
  description: 'description',
  exampleThumb: 'http://example.thumb.com',
  exampleUrl: 'http://example.url.com',
};

const mocks = [
  {
    request: {
      query: GET_PROJECTS_QUERY,
    },
    result: {
      data: {
        projects: [testProject],
      },
    },
  },
  {
    request: {
      query: GET_PROJECT_QUERY,
      variables: {
        projectId: testProject._id,
      },
    },
    result: {
      data: {
        project: testProject,
      },
    },
  },
];

describe('explore marketplace page', () => {
  test('should render loading state initially', () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ExploreMarketplacePage />
      </MockedProvider>,
    );
    expect(wrapper.find(Loading)).toBeDefined();
  });
});

describe('contact page', () => {
  test('should render loading state initially', () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ContactPage projectId={testProject._id} />
      </MockedProvider>,
    );
    expect(wrapper.find(Loading)).toBeDefined();
  });

  test('should match snapshot after query load', async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ContactPage projectId={testProject._id} />
      </MockedProvider>,
    );
    const tree = component.toJSON();
    await wait(0);
    expect(tree).toMatchSnapshot();
  });
});
