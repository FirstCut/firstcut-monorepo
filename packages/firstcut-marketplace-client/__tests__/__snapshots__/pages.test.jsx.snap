// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should match snapshot after query load 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <img
    className="ui image"
    src="/marketplace.png"
    style={
      Object {
        "height": "25px",
        "left": 0,
        "marginLeft": "20px",
        "marginTop": "20px",
        "position": "absolute",
        "top": 0,
      }
    }
  />
</div>
`;

exports[`contact page should match snapshot after query load 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="ui stackable computer reversed grid"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      align="center"
      className="eight wide computer sixteen wide mobile sixteen wide tablet column"
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <div
        style={
          Object {
            "maxWidth": "500px",
          }
        }
      >
        <div
          align="left"
          className="ui green header"
        >
          Title
        </div>
        <div
          className="ui embed"
          onClick={[Function]}
          style={
            Object {
              "marginBottom": "20px",
            }
          }
        >
          <i
            aria-hidden="true"
            className="video play icon"
          />
          <img
            className="placeholder"
            src="http://example.thumb.com"
          />
        </div>
        <i>
          description
        </i>
      </div>
    </div>
    <div
      align="center"
      className="eight wide computer sixteen wide mobile sixteen wide tablet column"
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <div
        style={
          Object {
            "maxWidth": "400px",
          }
        }
      >
        <div
          align="left"
          className="ui green header"
        >
          Contact us
        </div>
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <div
            className="equal width fields"
          >
            <div
              className="field"
            >
              <div
                className="required field"
              >
                <div
                  className="ui input"
                >
                  <input
                    name="firstName"
                    onChange={[Function]}
                    placeholder="First Name"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="field"
            >
              <div
                className="required field"
              >
                <div
                  className="ui input"
                >
                  <input
                    name="lastName"
                    onChange={[Function]}
                    placeholder="Last Name"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="required field"
            >
              <div
                className="ui input"
              >
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="required field"
            >
              <div
                className="ui input"
              >
                <input
                  name="company"
                  onChange={[Function]}
                  placeholder="Company Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  name="website"
                  onChange={[Function]}
                  placeholder="Company Website"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  name="location"
                  onChange={[Function]}
                  placeholder="Where would you like to shoot your video?"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="field"
            >
              <div
                className="ui input"
              >
                <input
                  name="budget"
                  onChange={[Function]}
                  placeholder="What is your estimated budget range?"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="field"
            >
              <textarea
                name="about"
                onChange={[Function]}
                onInput={[Function]}
                placeholder="Anything about this project you would like us to know before we contact you?"
                rows={3}
                style={
                  Object {
                    "resize": "",
                  }
                }
                value=""
              />
            </div>
          </div>
          <div
            className="field"
          >
            <button
              className="ui green fluid button"
              onClick={[Function]}
              role="button"
            >
              SUBMIT
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`explore marketplace page should match snapshot after query load 1`] = `
<div
  className="ui container"
  style={
    Object {
      "paddingTop": "100px",
    }
  }
>
  <div
    className="ui centered cards"
  />
</div>
`;
